build/source/netcdf/def_output.f90

```diff
@@ -59,71 +60,84 @@ module def_output_module
  subroutine def_output(nHRU,nSoil,infile,err,message)
  USE globalData,only:structInfo                               ! information on the data structures
- USE globalData,only:forc_meta,attr_meta,type_meta            ! metadata structures
- USE globalData,only:prog_meta,diag_meta,flux_meta,deriv_meta ! metadata structures
- USE globalData,only:mpar_meta,indx_meta                      ! metadata structures
- USE globalData,only:bpar_meta,bvar_meta                      ! metadata structures
+ USE globalData,only:forc_meta,attr_meta,type_meta            ! metaData structures
+ USE globalData,only:prog_meta,diag_meta,flux_meta,deriv_meta ! metaData structures
+ USE globalData,only:mpar_meta,indx_meta                      ! metaData structures
+ USE globalData,only:bpar_meta,bvar_meta,time_meta            ! metaData structures
  USE globalData,only:model_decisions
+ USE globalData,only:ncid
+ USE globalData,only:nFreq,outFreq                            ! output frequencies
  USE multiconst,only:integerMissing
  ! declare dummy variables
- integer(i4b), intent(in)    :: nHRU                          ! number of HRUs
- integer(i4b), intent(in)    :: nSoil                         ! number of soil layers in the first HRU (used to define fixed length dimensions)
- character(*), intent(in)    :: infile                        ! file suffix
+ integer(i4b),intent(in)     :: nHRU                          ! number of HRUs
+ integer(i4b),intent(in)     :: nSoil                         ! number of soil layers in the first HRU (used to define fixed length dimensions)
+ character(*),intent(in)     :: infile                        ! file suffix
  integer(i4b),intent(out)    :: err                           ! error code
  character(*),intent(out)    :: message                       ! error message
  ! local variables
- integer(i4b)                :: ivar                          ! loop through model variables
- integer(i4b)                :: iStruct                       ! index of data structure
+ integer(i4b)                :: iVar                          ! loop through model variables
+ integer(i4b)                :: iFreq                         ! loop through output frequencies
+ integer(i4b)                :: iStruct                       ! loop through structure types 
+ integer(i4b),parameter      :: modelTime=1                   ! model timestep output frequency
+ character(len=5)            :: fstring                       ! string to hold model output freuqnecy
+ character(len=1000)         :: fname                         ! temporary filename
  character(len=256)          :: cmessage                      ! temporary error message
+
  ! initialize errors
  err=0; message="def_output/"
 
  ! create initial file
- call ini_create(nHRU,nSoil,trim(infile),err,cmessage)
- if(err/=0)then; message=trim(message)//trim(cmessage); return; endif
+ do iFreq = 1,nFreq
+  write(fstring,'(i5)') outFreq(iFreq)
+  fstring = adjustl(fstring)
```

Comment:
Can you convert `fString` to something more meaningful than units of time steps? For example, in units of hours, days or months (whatever is closest).